player/player.go

```diff
@@ -0,0 +1,227 @@
+	if len(m) == 0 {
+		return 0, 0
+	}
+	start, err1 := strconv.ParseInt(m[1], 10, 64)
+	end, err2 := strconv.ParseInt(m[2], 10, 64)
+	if err1 != nil || err2 != nil {
+		return 0, 0
+	}
+	if start < 0 {
+		start = 0
+	}
+	if end < 0 {
+		end = 0
+	}
+	if start > end {
+		start = 0
+		end = 0
+	}
+	return start, end
+}
+
+func newReflectedStream(uri string) *reflectedStream {
+	return &reflectedStream{URI: uri}
+}
+
+func (s *reflectedStream) URL() string {
+	return reflectorURL + string(s.SDHash)
+}
+
+// resolve gets SD hash from the daemon for a given URI to enable further retrieval of stream content from
+// reflector servers
+func (s *reflectedStream) resolve() error {
+	client := ljsonrpc.NewClient(config.Settings.GetString("Lbrynet"))
+	response, err := client.Resolve(s.URI)
+	if err != nil {
+		return err
+	}
+	source := (*response)[s.URI].Claim.Value.Stream.Source
+	s.SDHash = source.Source
+	s.ContentType = source.GetContentType()
+	monitor.Logger.WithFields(log.Fields{
+		"sd_hash":      fmt.Sprintf("%s", s.SDHash),
+		"uri":          s.URI,
+		"content_type": s.ContentType,
+	}).Info("resolved uri")
+	return nil
+}
+
+// fetchData gets stream data from the reflector and does some blob magic along the way
+func (s *reflectedStream) fetchData() error {
+	if s.SDHash == nil {
+		return errors.Err("No hash set, call `resolve` first")
+	}
+	monitor.Logger.WithFields(log.Fields{
+		"uri": s.URI,
+		"url": s.URL(),
+	}).Info("requesting stream data")
+	httpRequest, err := http.NewRequest("GET", s.URL(), new(bytes.Buffer))
+	httpClient := &http.Client{}
+	httpResponse, err := httpClient.Do(httpRequest)
+	if err != nil {
+		return err
+	}
+	defer httpResponse.Body.Close()
+	err = json.NewDecoder(httpResponse.Body).Decode(&s)
+	if err != nil {
+		return err
+	}
+	decodedKey, err := hex.DecodeString(s.Key)
+	if err != nil {
+		return err
+	}
+	s.DecodedKey = decodedKey
+	var blobsSizes string
+	for n, blob := range s.Blobs {
+		if blob.Length == stream.MaxBlobSize {
+			s.Size += blob.Length - 1
+		} else {
+			s.Size += blob.Length
+		}
+		decodedIV, err := hex.DecodeString(blob.IV)
+		if err != nil {
+			return err
+		}
+		s.Blobs[n].DecodedIV = decodedIV
+		if blobsSizes == "" {
+			blobsSizes = fmt.Sprintf("%v", blob.Length)
+		} else {
+			blobsSizes = fmt.Sprintf("%v+%v", blobsSizes, blob.Length)
+		}
+	}
+	// last padding is unguessable
+	s.Size -= 15
+	sort.Slice(s.Blobs, func(i, j int) bool { return s.Blobs[i].Number < s.Blobs[j].Number })
+	monitor.Logger.WithFields(log.Fields{
+		"blobs_number": len(s.Blobs),
+		"blobs_sizes":  blobsSizes,
+		"stream_size":  s.Size,
+		"uri":          s.URI,
+	}).Info("got stream data")
+	return nil
+}
+
+// prepareWriter writes necessary range headers and sets the status, preparing HTTP for partial blob streaming.
+// This should be called before streamBlobs
+func (s *reflectedStream) prepareWriter(startByte, endByte int64, writer http.ResponseWriter) (startBlob, endBlob int) {
```

Comment:
this function does two separate things: calculate the byte ranges and write headers. they should not be tied together.